### PR TITLE
Leverage Functional Tests

### PR DESCRIPTION
**Pull request by rbajollari**
Tests to implement:

- [x] https://github.com/umee-network/umee/issues/1307
- [x] https://github.com/umee-network/umee/issues/1210

First test that does not require price-feeder to be running has been completed and should be ready to merge in after a review. Its separated into two unit tests (test_supply_withdraw testing with only uumee and test_supply_withdraw_atom testing with ibc/atom). 

Second test with borrow (requiring price-feeder to run) is finished and ready for review

### CODE REVIEW

**Review by toteki — APPROVED** [review]
Looks solid.

Note that I haven't been able to test on OSX yet (applies to the repository as a whole), so this is an untested review.

For later scenarios, emphasis might be less on testing that an individual user's balance increased/decreased by the right amount after one transaction, but that the transaction can be run 10000x by a bunch of users without failing, for example. In those tests, the `self.assertEqual` lines won't be needed.

**Comment by khoerling** [comment]
@toteki welcome to give this PR a spin, `make setup-chain` should work on osx now:  https://github.com/umee-network/qa-integration/pull/12

**Comment by adamewozniak** [comment]
@rbajollari is this ready to be merged in?

**Comment by rbajollari** [comment]
@adamewozniak Withdraw leverage tx is not working currently in my tests after all the recent updates which is causing those first two tests which were passing before hand to not anymore. I'm currently trying to figure out whats causing that but once I do this branch should be ready to merge in.

Also I'll create separate prs for the other tests.

**Comment by adamewozniak** [comment]
> 

Right on, if you need a second hand lmk 😄

File: internal/modules/leverage/test.py [hunk 1]
```diff
@@ -3,35 +3,445 @@
...
 )
+from modules.leverage.tx import (
+    tx_supply,
+    tx_withdraw,
+    tx_collateralize,
+    tx_borrow,
+    tx_repay,
+)
+
+BLOCKS_PER_VOTING_PERIOD = 10
```
Comment (adamewozniak) [inline internal/modules/leverage/test.py:46]: Did this break any of the tests @zarazan put in for the oracle?

File: internal/modules/leverage/test.py [hunk 2]
```diff
@@ -3,35 +3,445 @@
...
+            status = tx_withdraw(accounts[0]["name"], "1000000u/uumee", validator1_home)
+            self.assertTrue(status)
 
     def test_query_total_supply(self):
         status, res = query_registered_tokens()
         self.assertTrue(status)
-        self.assertTrue(len(res['registry']) >= 1, "It should have at least one token registered")
+        self.assertTrue(len(res['registry']) == 3, "It should have three tokens registered")
+
+    def test_supply_withdraw(self):
```
Comment (adamewozniak) [inline internal/modules/leverage/test.py:143]: I'd add a ref here to the GH issue describing this test

File: internal/modules/leverage/test.py [hunk 3]
```diff
@@ -3,35 +3,445 @@
...
+
+    # def test_functional_one(self):
+    #     # Submit exhange rates to price feeder every voting period in the background
+    #     stop_exchange_rate_set = False
+    #     exchange_rate_set_thread = threading.Thread(target=self.exchange_rate_set, args=(EXCHANGE_RATES, ))
+    #     exchange_rate_set_thread.start()
+    #     wait_for_next_voting_period()
+
+    #     # account1, ..., account50 supply and collateralize 10000 umee
+    #     for i in range(50):
```
Comment (adamewozniak) [inline internal/modules/leverage/test.py:284]: Please delete this commented code

File: internal/modules/oracle/query.py [hunk 4]
```diff
@@ -1,8 +1,10 @@
+import time
 from utils import exec_command, env
 
 DAEMON = env.DAEMON
 RPC = env.RPC
 CHAINID = env.CHAINID
+BLOCKS_PER_VOTING_PERIOD = 5
```
Comment (adamewozniak) [inline internal/modules/oracle/query.py:7]: Should this be updated too?

File: scripts/chain/start_chain.sh [hunk 5]
```diff
@@ -167,7 +178,8 @@ jq '.app_state["gravity"]["params"]["bridge_ethereum_address"]="0x93b5122922F9dC
     | .app_state["gravity"]["params"]["bridge_active"]=false
     | .app_state["gravity"]["delegate_keys"]=[{"validator":"umeevaloper1y6xz2ggfc0pcsmyjlekh0j9pxh6hk87ymuzzdn","orchestrator":"umee1y6xz2ggfc0pcsmyjlekh0j9pxh6hk87ymc9due","eth_address":"0xfac5EC50BdfbB803f5cFc9BF0A0C2f52aDE5b6dd"},{"validator":"umeevaloper1qjehhqdnc4mevtsumk6nkhm39nqrqtcy2f5k6k","orchestrator":"umee1qjehhqdnc4mevtsumk6nkhm39nqrqtcy2dnetu","eth_address":"0x02fa1b44e2EF8436e6f35D5F56607769c658c225"},{"validator":"umeevaloper1s824eseh42ndyawx702gwcwjqn43u89dhmqdw8","orchestrator":"umee1s824eseh42ndyawx702gwcwjqn43u89dhl8zld","eth_address":"0xd8f468c1B719cc2d50eB1E3A55cFcb60e23758CD"}]
     | .app_state["gravity"]["gravity_nonces"]["latest_valset_nonce"]="0"
-    | .app_state["gravity"]["gravity_nonces"]["last_observed_nonce"]="0"' \
+    | .app_state["gravity"]["gravity_nonces"]["last_observed_nonce"]="0" 
+    | .app_state["oracle"]["params"]["vote_period"] = "10"'\
```
Comment (adamewozniak) [inline scripts/chain/start_chain.sh:182]: Add a comment here for why we changed this please

**Review by adamewozniak — CHANGES_REQUESTED** [review]
Just a few suggestions - please also merge main into this branch to make the diff smaller

**Review by adamewozniak — APPROVED** [review]
lgtm! @rbajollari please resolve conflicts & merge

**Pull request merged**